proposal.html

```diff
@@ -556,7 +556,7 @@ <h1>(<emu-xref href="#sec-function.prototype.tostring">19.2.3.5</emu-xref>) Func
   <ins class="block">
     <emu-alg>
       1. If _func_ is a <emu-xref href="#sec-bound-function-exotic-objects">Bound Function exotic object</emu-xref> or a <emu-xref href="#sec-ecmascript-standard-built-in-objects">built-in Function object</emu-xref>, then return an implementation-dependent String source code representation of _func_. The representation must have the syntax of a |NativeFunction|. Additionally, if _func_ is a <emu-xref href="#sec-well-known-intrinsic-objects">Well-known Intrinsic Object</emu-xref>, the portion of the returned String that would be matched by |IdentifierName| must be the initial value of the *name* property of _func_.
-      1. If _func_ has a [[SourceText]] internal slot and Type(_func_.[[SourceText]]) is String and ? HostHasSourceTextAvailable(_func_), then return _func_.[[SourceText]].
+      1. If _func_ has a [[SourceText]] internal slot and Type(_func_.[[SourceText]]) is String and HostHasSourceTextAvailable(_func_) is *true*, then return _func_.[[SourceText]].
```

Comment:
should this have a `!` since it won't ever be abrupt?

Reply:
Hah, this comes back to the issue @jmdyck raised--it depends whether you believe all abstract operations return a completion record!

Reply:
Well this one certainly does.

Reply:
> it depends whether you believe all abstract operations return a completion record!

(It's actually orthogonal to that.)

Reply:
Well, HostHasSourceTextAvailable returns a completion record, I agree with @ljharb that a `!` would be good.